docs/uri-formattering.md

```diff
@@ -0,0 +1,61 @@
+# Uri formattering
+
+In Enterprise systemen is het gebruikelijk om urls in relatieve en absolute formaat te werken.
+
+Absolute urls zijn handig voor applicaties die rechtstreeks verbinding kunnen maken met Web APIs. Voorbeelden van zulke applicaties zijn desktop applicaties, web applicaties die deels of geheel op de server draaien en batch applicaties. Het belangrijkste voordeel van absolute urls is dat ze zonder transformatie kunnen worden gebruikt.
+
+Moderne web applicaties die volledig in een browser draaien (ook bekend als Single Page Applications) kunnen standaard geen absolute urls gebruiken. Om security redenen implementeren alle browsers de [Same-origin Policy](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy) waardoor code die wordt gedraaid in een browser geen aanroepen kan sturen naar resources van een ander [origin](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy#definition_of_an_origin).
+Om het voor een moderne web applicatie mogelijk te maken aanroepen te sturen naar Web APIs van een ander origin, is het een best practice om de Backend for Frontend (BFF) pattern te implementeren. De Backend is een applicatie die op de server draait om de aanroepen van de Frontend te routeren naar Web APIs.
+
+Web APIs die de HATEOAS constraint implementeren, leveren in hun responses ook urls om de volgende mogelijke aanroepen naar hun consumers te communiceren. Deze aanroepen zijn meestal aanroepen naar resources die de Web API zelf beheert en kunnen daarom relatieve urls zijn. Een belangrijk voordeel van het gebruik van relatieve urls door een Web API is schaalbaarheid. Meerdere instanties van zo'n Web API kan worden opgestart zonder voor elke instantie url configuratie aanpassingen te hoeven maken. Dit vergemakkelijkt ook het deployen van de Web API naar verschillende omgevingen (Test, Acceptatie, Productie). Dit is echter niet mogelijk als de response van een Web API ook urls kan bevatten naar resources van externe Web APIs.
```

Comment:
Corrigeer
> Meerdere instanties van zo'n Web API kan worden opgestart zonder ...

in

_Meerdere instanties van zo'n Web API **kunnen** worden opgestart zonder ..._

en
> ... als de response van een Web API ook urls kan bevatten ...

in

_... als de response van een Web API ook urls **kunnen** bevatten ..._

Ik zoek overigens nog naar reden van het opnemen van de laatste zin in deze alinea. 

Ik kan me er wat bij voorstellen dat, als je urls naar externe resources opneemt, deployen naar verschillende omgevingen niet kan maar daar biedt de in dit document geschetste uri formattering toch geen oplossing voor of mis ik dan iets.
Het doel van dit document is duidelijk uit te leggen wat uri formatering is en hoe dat werkt. Voor mij is de laatste zin dan ineens een opmerking die daar niets aan bijdraagt en me zelfs afleidt. Tenzij je het daarin geschetste probleem ook met uri formattering op kunt lossen.

Reply:
Ja, dat kan. Ik zal kijken hoe ik dat kan verduidelijken/uitleggen